src/Aspire.Hosting.AWS/Provisioning/Provisioners/CloudFormationResourceProvisioner.cs

```diff
@@ -0,0 +1,127 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using Amazon.CloudFormation;
+using Amazon.CloudFormation.Model;
+using Amazon.Runtime;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.AWS.CloudFormation;
+using Aspire.Hosting.AWS.Provisioning.Provisioners;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Hosting.AWS.Provisioning;
+
+internal abstract class CloudFormationResourceProvisioner<T>(ResourceLoggerService loggerService, ResourceNotificationService notificationService) : AWSResourceProvisioner<T>
+    where T : ICloudFormationResource
+{
+
+    protected ResourceLoggerService LoggerService => loggerService;
+
+    protected ResourceNotificationService NotificationService => notificationService;
+
+    protected virtual Task<CloudFormationStackExecutionContext> CreateCloudFormationExecutionContext(T resource, CancellationToken cancellationToken)
+        => throw new NotImplementedException();
+
+    protected async Task ProvisionCloudFormationTemplateAsync(T resource, CancellationToken cancellationToken)
```

Comment:
Shouldn't this be in `CloudFormationTemplateResourceProvisioner`?

Reply:
Will replace this

Reply:
The `IStackResource` is not a `ICloudFormationTemplateResource` because this needs a `TemplatePath`. The common denominator is the `ICloudFormationResource`